### ISSUE TITLE
Separate executions sometimes swap environment variables

### ISSUE THREAD

**Issue by royjenkins** [rundeck/rundeck#1330]
Hi,

I have found that there is a race condition when executing two or more jobs across multiple remote nodes in parallel. It seems that both executions are sharing the same environment, which in some instances causes another executions environment variables to be exported in place of the correct ones.

This can easily be replicates using the following job definition below, which simply prints out the execution ID on each node. I fired these jobs of rapidly using the CLI:

    run -p <myProject> -f -j crossed-wires &
    run -p <myProject> -f -j crossed-wires

In my resulting log, I consistently find the execution ID for the second execution mixed in the first log one or more times.

    <joblist>
      <job>
        <id>f716e039-f652-42a9-8ad1-413b139bc489</id>
        <loglevel>INFO</loglevel>
        <multipleExecutions>true</multipleExecutions>
        <sequence keepgoing='false' strategy='node-first'>
          <command>
            <scriptargs />
            <script><![CDATA[#!/bin/bash

    echo @job.execid@]]></script>
          </command>
        </sequence>
        <description></description>
        <name>crossed-wires</name>
        <context>
          <project>Zynstra</project>
        </context>
        <dispatch>
          <threadcount>1</threadcount>
          <keepgoing>false</keepgoing>
          <excludePrecedence>true</excludePrecedence>
          <rankOrder>ascending</rankOrder>
        </dispatch>
        <uuid>f716e039-f652-42a9-8ad1-413b139bc489</uuid>
        <nodefilters>
          <filter>osFamily:unix</filter>
        </nodefilters>
      </job>
    </joblist>

Unfortunately, we are currently stuck at 2.4.2-1 due to another issue, so I have not been able to confirm that this is still present in the latest release.

**Comment by gschueler** [comment]
see #1129 this was fixed in 2.5.0

**Comment by royjenkins** [comment]
I found somewhere to install it and can confirm this problem is not present in 2.5.0. Thanks.
